apps/prairielearn/src/pages/instructorLoadFromDisk/instructorLoadFromDisk.ts

```diff
@@ -23,7 +27,13 @@ async function update(locals: Record<string, any>) {
 
   serverJob.executeInBackground(async (job) => {
     let anyCourseHadJsonErrors = false;
-    await async.eachOfSeries(config.courseDirs || [], async (courseDir, index) => {
+
+    // Merge the list of courses in the config with the list of courses in the database.
+    const courseDirs = new Set<string>(config.courseDirs);
+    const courses = await queryRows(sql.select_all_courses, CourseSchema);
+    courses.forEach((course) => courseDirs.add(course.path));
```

Comment:
Use concat?

Reply:
`courseDirs` is (intentionally) a `Set`, which doesn't have a `concat` method.

Reply:
I missed that... Union might be an option then, though I'm ok with the code as is. That said, I think it's important to keep insertion order, as I keep the directories in the config in insertion order to ensure courses I'm most interested in are synced faster. The current implementation does that.

Reply:
[`Set.prototype.union()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Set/union) is only available for Node 22. We're running on Node 20.

Reply:
I called out that we deliberately maintain insertion order in an updated comment in 92ab240.